opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -624,6 +624,11 @@ const std::vector<std::shared_ptr<Dune::cpgrid::CpGridData>>& CpGrid::chooseData
     }
 }
 
+std::vector<std::shared_ptr<Dune::cpgrid::CpGridData>>& CpGrid::getData()
+{
+    return !distributed_data_.empty() ? distributed_data_ : data_;
```

Comment:
The decision should be based on what current_view_data_ points to.
```suggestion
    return current_view_data_ == data_.back() ? data_ : distributed_data_;
```